overwatch/src/services/resources.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{
+    overwatch::handle::OverwatchHandle,
+    services::{
+        life_cycle::LifecycleHandle, relay::InboundRelay, settings::SettingsUpdater,
+        state::StateUpdater, status::StatusHandle,
+    },
+};
+
+/// Core resources for a `Service`.
+///
+/// Contains everything required to start a new
+/// [`ServiceRunner`](crate::services::runner::ServiceRunner).
+pub struct ServiceResources<Settings, State, RuntimeServiceId> {
```

Comment:
Why can't we put everything under `ServiceRunner` instead? If this type serves no other purpose (it does not as far as I can see), it would make it easier to understand code if it would all be grouped under `ServiceRunner` instead.

Also because, all of these fields are also part of a `ServiceRunner`, so I am not sure why we need to duplicate these fields? Clarification would be amazing 😃

Reply:
This is hold by the runner now and given to the service itself when started. But is kept after the service is killed. If I understood the thing properly.

Reply:
My point is that there seems to be clones of the same objects in multiple places, and I was trying to figure out if that is what we need. My [comment above](https://github.com/logos-co/Overwatch/pull/79#discussion_r2088166613) was one of such instances, where the `overwatch_handle` is present in all objects, even the ones that do not care about such handle. But this could be because I still have not grasped 100% of the design intentions here.

Reply:
Having this entity and having these attributes as `ServiceRunner`'s attributes is effectively the same, as this is held by the `ServiceRunner`. 
They're just grouped to give context to them, which I think is a good thing.

Reply:
Yeah my point is that `ServiceRunner` has all those properties, plus a property which contains again all those properties, right? I like the reason of grouping them, but grouping could also be done only when needed by a caller, by pulling from the runner state. This is, of course, if I did not misunderstand anything in the current design.

Reply:
Oh, that's right. I didn't realise they were duplicated 🤦‍♀️
Good catch, will prune them off.

Reply:
Oh, indeed. I didn't realise they were duplicated.
I've done an attempt on some attribute juggling. I have mixed feelings about it: It allows for a clean syntax, but I have a an "off feeling" about it.

Reply:
What don't you like about it?

Reply:
Probably the handling of the `inbound_relay`. 
It's not hacky code, but it doesn't feel clearly intuitive. Although, to be fair, it's kind of expected as that whole behaviour (retrieving it from `Service`, etc.) is not obvious in the first place.
If you guys are on board, so am I :)